test/suites/integration/subnet_test.go

```diff
@@ -54,6 +54,32 @@ var _ = Describe("Subnets", func() {
 
 		env.ExpectInstance(pod.Spec.NodeName).To(HaveField("SubnetId", HaveValue(Equal(firstSubnet))))
 	})
+	It("should use the subnet-id selector with resource based naming", func() {
+		subnets := getSubnets(map[string]string{"karpenter.sh/discovery": settings.FromContext(env.Context).ClusterName})
+		Expect(len(subnets)).ToNot(Equal(0))
+		shuffledAZs := lo.Shuffle(lo.Keys(subnets))
+		firstSubnet := subnets[shuffledAZs[0]][0]
+
+		enableResourceNameDNSArecord(firstSubnet)
```

Comment:
instead of picking one subnet, can we configure all the subnets with the selector to use resource based names? This will help to prevent flakey tests when there are operational issues occurring in one AZ. 

You can change the func signature of `enableResourceNameDNSArecord` to a variadic: 

```
func enableResourceNameDNSArecord(subnetIDs ...string) {
```